### PR TITLE
feat: use networkClientId to resolve chainId in PPOM Middleware

### PR DESCRIPTION
**Pull request by jiexi**
## **Description**

Updates the PPOM Middleware to use the chainId for the dapp selected network instead of the globally selected network when processing a request

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27263?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/lib/ppom/ppom-middleware.ts
Comment (adonesky1) [inline app/scripts/lib/ppom/ppom-middleware.ts:86]: worth mentioning that `SelectedNetworkController` middleware (and the middleware to tag on the networkClientId in the full multichain API now **has to** come before this middleware)

File: app/scripts/lib/ppom/ppom-middleware.ts
Comment (adonesky1) [inline app/scripts/lib/ppom/ppom-middleware.ts:90]: what is the bang for here again?

Comment (adonesky1) [reply app/scripts/lib/ppom/ppom-middleware.ts:90]: Also do we need some error handling in case this request fails somehow?

Comment (jiexi) [reply app/scripts/lib/ppom/ppom-middleware.ts:90]: bang tells typescript that this result will always be defined (vs `undefined`). If the SelectedNetworkMiddleware puts an invalid networkClientId on the request, I think it's probably fine if this blows up trying to dereference chainId since we have bigger issues. 

If we were to do error handling, it seems like we should throw earlier in the middleware stack? Perhaps in the SelectedNetworkMiddleware itself?

File: app/scripts/lib/ppom/ppom-middleware.ts
Comment (jiexi) [reply app/scripts/lib/ppom/ppom-middleware.ts:86]: added comment to MetaMaskController here https://github.com/MetaMask/metamask-extension/pull/27263/commits/d30b4a53108a221e71b122a2450ffd413e74cf0e

**Review by adonesky1 — APPROVED** [review]
Seems correct to me but we should definitely get an approval from @jpuri or someone else on confirmations before merging!

**Review by jpuri — APPROVED** [review]

**Comment by jiexi** [comment]
tested this by trying to send a malicious tx on a different chain from the globally selected by setting `QueuedRequestController.shouldRequestSwitchNetwork` to always return false. Verified in the PPOM controller that the request's chainId was different from the globally selected chain Id. Got the PPOM warning in the confirmation as expected

To be extra safe, the globally selected network was set to Linea Sepolia, a chain that PPOM does not support. This implies that the warning shown must have used the request's chainId

**Pull request merged**
